homeassistant/components/advantage_air/__init__.py

```diff
@@ -60,7 +61,7 @@ async def error_handle(param):
                 if await func(param):
                     await coordinator.async_refresh()
             except ApiError as err:
-                _LOGGER.warning(err)
+                raise HomeAssistantError(err) from err
```

Comment:
I think you could add type hints to `async_get`, `error_handle_factory` and `error_handle` here within `async_setup_entry`.

Reply:
I've added what I can, but the underlying library is not typed (yet), and I'm still getting to grips with typing.